docs/api/utils/selfDependant.md

```diff
@@ -2,25 +2,40 @@
 title: selfDependant()
 ---
 
-A creation operator that helps creating observables that have circular
-dependencies.
+An RxJS creation operator for observables that have circular dependencies.
+
+```ts
+function selfDependant<T>(): [Observable<T>, () => MonoTypeOperatorFunction<T>];
+```
+
+#### Returns
+
+`[1, 2]`:
+
+1. An Observable... (**TODO**)
+
+2. A pipeable operator... (**TODO**)
```

Comment:
This is also tough to explain :disappointed:  And we should really come up with a better example. However, for now I would stick to what we have in the JSDocs:

1. The inner-subject as an Observable
2. A pipeable operator that taps into the inner Subject

Reply:
What do you mean by "the inner subject"?

Reply:
Ok, let's see if I manage to explain this a bit.

The thing is that sometimes we have observables that have circular dependencies. For instance, consider the shitty example of that's in the docs of this operator:

We want to create this `resettableCounter$` observable, which is like a normal counter, but that if no one has clicked on the counter for a while it resets its value to zero. This problem can be modeled in other ways that don't imply a circular dependency, that's why I don't love this example, but just for arguments sake, lets say that we want to model it like this:

We want to have an `inc$` observable that whenever a user clicks to it it emits the next increment... However, in order to do that it needs to know what the latest value was and the latest value comes out of the `resettableCounter$`... So, if model this problem like this, then we have a circular dependency, how could we solve this problem without `selfDependant`? Like this:

```ts
import { merge, of, Subject } from 'rxjs'
import { delay, map, share, switchMapTo, tap, withLatestFrom } from 'rxjs/operators'
import { selfDependant } from '@react-rxjs/utils'

const _resettableCounter$ = new Subject<number>()
const clicks$ = new Subject()

const inc$ = clicks$.pipe(
  withLatestFrom(_resettableCounter$),
  map((_, x) => x + 1),
  share(),
)
const delayedZero$ = of(0).pipe(delay(10_000))
const reset$ = inc$.pipe(switchMapTo(delayedZero$))
const resetableCounter$ = merge(inc$, reset$, of(0)).pipe(
  connectResetableCounter(),
const resettableCounter$ = merge(inc$, reset$, of(0)).pipe(
  tap(_resettableCounter),
)
```

That's it, that's all what `selfDependant` does.

In fact, I initially decided against adding this helper... However, later on I changed my mind [because of the following reasons](https://github.com/re-rxjs/react-rxjs/pull/75#issuecomment-681847695):

- This is a more declarative way of defining cyclical/recursive Observables. In the moment of its definition you already tell what that is about.

- This is a pattern that's difficult to get right and that it's worth promoting. For instance: trying to use a `withLatestFrom` with a `defer` returning the self-dependant observable is a tempting approach, but it's wrong because it generates a subscription that can't be teared down.

- In the future, using this operator can make the integration with the dev-tools (rxjs-traces) easier (on a development build).

What I'm trying to say is that **this helper is purposely leaky**, because it's just a more declarative/safer way of creating a Subject and tapping into it because of a circular dependency, nothing else.

If you can help me to explain that stuff better, I would really appreciate that.